File: CourseWorkOnAndroidAndSpring-main/project/server/src/main/java/ru/ryabov/studentperformance/dto/user/CreateStudentRequest.java
package ru.ryabov.studentperformance.dto.user;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public class CreateStudentRequest {

    @NotBlank(message = "Login is required")
    @Size(max = 50)
    private String login;

    @Size(max = 255)
    private String password;

    @NotBlank(message = "Email is required")
    @Size(max = 100)
    private String email;

    @NotBlank(message = "Last name is required")
    @Size(max = 50)
    private String lastName;

    @NotBlank(message = "First name is required")
    @Size(max = 50)
    private String firstName;

    @Size(max = 50)
    private String middleName;

    private Long groupId;
    @Size(max = 20)
    private String recordBookNumber;
    private Integer admissionYear;

    public String getLogin() { return login; }
    public void setLogin(String login) { this.login = login; }

    public String getPassword() { return password; }
    public void setPassword(String password) { this.password = password; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getLastName() { return lastName; }
    public void setLastName(String lastName) { this.lastName = lastName; }

    public String getFirstName() { return firstName; }
    public void setFirstName(String firstName) { this.firstName = firstName; }

    public String getMiddleName() { return middleName; }
    public void setMiddleName(String middleName) { this.middleName = middleName; }

    public Long getGroupId() { return groupId; }
    public void setGroupId(Long groupId) { this.groupId = groupId; }

    public String getRecordBookNumber() { return recordBookNumber; }
    public void setRecordBookNumber(String recordBookNumber) { this.recordBookNumber = recordBookNumber; }

    public Integer getAdmissionYear() { return admissionYear; }
    public void setAdmissionYear(Integer admissionYear) { this.admissionYear = admissionYear; }
}

File: CourseWorkOnAndroidAndSpring-main/project/mobile/app/src/main/java/ru/zagrebin/studentperformance/data/remote/dto/CreateStudentRequest.kt
package ru.ryabov.studentperformance.data.remote.dto

import com.google.gson.annotations.SerializedName

data class CreateStudentRequest(
    @SerializedName("login") val login: String,
    @SerializedName("password") val password: String? = null,
    @SerializedName("email") val email: String,
    @SerializedName("lastName") val lastName: String,
    @SerializedName("firstName") val firstName: String,
    @SerializedName("middleName") val middleName: String? = null,
    @SerializedName("groupId") val groupId: Long? = null,
    @SerializedName("recordBookNumber") val recordBookNumber: String? = null,
    @SerializedName("admissionYear") val admissionYear: Int? = null
)
